extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/security/SecurityConfigFilter.java

```diff
@@ -0,0 +1,86 @@
+        this.config = config;
+    }
+
+    @Override
+    public void filterOpenAPI(OpenAPI openAPI) {
+
+        if (config.securityScheme.isPresent()) {
+
+            // Make sure components are created
+            if (openAPI.getComponents() == null) {
+                openAPI.setComponents(OASFactory.createComponents());
+            }
+
+            Map<String, SecurityScheme> securitySchemes = new HashMap();
+
+            // Add any existing security
+            if (openAPI.getComponents().getSecuritySchemes() != null
+                    && !openAPI.getComponents().getSecuritySchemes().isEmpty()) {
+                securitySchemes.putAll(openAPI.getComponents().getSecuritySchemes());
+            }
+
+            SmallRyeOpenApiConfig.SecurityScheme securitySchemeOption = config.securityScheme.get();
+
+            SecurityScheme securityScheme = OASFactory.createSecurityScheme();
+            securityScheme.setDescription(config.securitySchemeDescription);
+
+            switch (securitySchemeOption) {
+                case basic:
+                    securityScheme.setType(SecurityScheme.Type.HTTP);
+                    securityScheme.setScheme(config.basicSecuritySchemeValue);
+                    break;
+                case jwt:
+                    securityScheme.setType(SecurityScheme.Type.HTTP);
+                    securityScheme.setScheme(config.jwtSecuritySchemeValue);
+                    securityScheme.setBearerFormat(config.jwtBearerFormat);
+                    break;
+                case oidc:
+                    securityScheme.setType(SecurityScheme.Type.OPENIDCONNECT);
+                    securityScheme.setOpenIdConnectUrl(config.oidcOpenIdConnectUrl.orElse(null));
+                    break;
+                case oauth2Implicit:
+                    securityScheme.setType(SecurityScheme.Type.OAUTH2);
+                    OAuthFlows oAuthFlows = OASFactory.createOAuthFlows();
+                    OAuthFlow oAuthFlow = OASFactory.createOAuthFlow();
+                    if (config.oauth2ImplicitAuthorizationUrl.isPresent()) {
+                        oAuthFlow.authorizationUrl(config.oauth2ImplicitAuthorizationUrl.get());
+                    }
+                    if (config.oauth2ImplicitRefreshUrl.isPresent()) {
+                        oAuthFlow.authorizationUrl(config.oauth2ImplicitRefreshUrl.get());
+                    }
+                    if (config.oauth2ImplicitTokenUrl.isPresent()) {
+                        oAuthFlow.tokenUrl(config.oauth2ImplicitTokenUrl.get());
+                    }
+                    oAuthFlow.setScopes(OASFactory.createScopes());
+                    oAuthFlows.setImplicit(oAuthFlow);
+                    securityScheme.setType(SecurityScheme.Type.OAUTH2);
+                    securityScheme.setFlows(oAuthFlows);
+                    break;
+            }
+            securitySchemes.put(config.securitySchemeName, securityScheme);
```

Comment:
@phillip-kruger IMHO it would make sense to log a debug message here if more than one scheme is available (if some have come via the open api doc) that only a single scheme will be supported.

Reply:
Ok cool. So if the use has defined one with Annotations and in the config for example ? Maybe WARN rather then DEBUG ?

Reply:
Fixed